django/db/models/options.py

```diff
@@ -549,6 +541,15 @@ def get_field(self, field_name, many_to_many=None):
             # field map.
             return self.fields_map[field_name]
```

Comment:
Unfortunately these is an issue here! fields_map calls relation_tree, that is a cached property. If any of these two raise an exception (due to premature relation tree) or cache less information then cached_property will not work correctly.